renderdoc/driver/gl/wrappers/gl_emulated.cpp

```diff
@@ -1218,6 +1218,70 @@ void APIENTRY _glClearBufferData(GLenum target, GLenum internalformat, GLenum fo
 
 #pragma endregion
 
+#pragma region GLES Compatibility
+
+void APIENTRY _glGetTexImage(GLenum target, GLint level, GLenum format, GLenum type, void *pixels)
+{
+  if((format == eGL_DEPTH_COMPONENT && !HasExt[NV_read_depth]) ||
+     (format == eGL_STENCIL && !HasExt[NV_read_stencil]) ||
+     (format == eGL_DEPTH_STENCIL && !HasExt[NV_read_depth_stencil]))
+  {
+    // return silently, check was made during startup
+    return;
+  }
+
+  GLint width = 0, height = 0, depth = 0;
+  internalGL->glGetTexLevelParameteriv(target, level, eGL_TEXTURE_WIDTH, &width);
+  internalGL->glGetTexLevelParameteriv(target, level, eGL_TEXTURE_HEIGHT, &height);
+  internalGL->glGetTexLevelParameteriv(target, level, eGL_TEXTURE_DEPTH, &depth);
+
+  GLint texture = 0;
+  internalGL->glGetIntegerv(TextureBinding(target), (GLint *)&texture);
+
+  GLenum attachment = eGL_COLOR_ATTACHMENT0;
+  if(format == eGL_DEPTH_COMPONENT)
+    attachment = eGL_DEPTH_ATTACHMENT;
+  else if(format == eGL_STENCIL)
+    attachment = eGL_STENCIL_ATTACHMENT;
+  else if(format == eGL_DEPTH_STENCIL)
+    attachment = eGL_DEPTH_STENCIL_ATTACHMENT;
+
+  GLuint fbo = 0;
+  internalGL->glGenFramebuffers(1, &fbo);
+
+  PushPopFramebuffer(eGL_FRAMEBUFFER, fbo);
+
+  size_t sliceSize = GetByteSize(width, height, 1, format, type);
+  GLenum binding = TextureTarget(target);
```

Comment:
This call to `TextureTarget` seems weird since it's already a target not a binding. I thought it was redundant at first, but I guess this is so that all the cubemap face targets get flattened to `eGL_TEXTURE_2D`?

I'd prefer to just add the 6 face enums into the case statement below, but you could also use the `TextureBinding` and switch on that enum (since it's mostly 1:1 apart from that flattening). At least then it makes more sense when reading the code.

This also raises another point - Buffer texture, and 1D/1D array textures can't be handled like this. Buffer we could always special-case to look up the underlying buffer and return that, but 1D would need to be treated like unsupported depth/stencil textures as above and somehow going through a slow path to copy them. For now just make sure the function will handle that case and early-out instead of making invalid GL calls.